README.md

```diff
@@ -237,19 +237,43 @@ class Calendar(component.Component):
 
 And voilá!! We've created our first component.
 
-## Autodiscovery
+## Using single-file components
```

Comment:
Moved the section on "single-file components" and "components as Views" higher up. Might not be easy to see in this diff, and just my personal opinion, but it feels to me that now README flows even a bit better:
1. installation
2. creating components (and the different variants thereof)
3. registering componnt with autodiscovery
4. using slots and passing data to components (which requires the other component to be registered)
5. and them some more niche stuff

Overall the ordering is:

<img width="546" alt="Screenshot 2024-05-13 at 16 19 07" src="https://github.com/EmilStenstrom/django-components/assets/25986782/4fa015d7-4cd3-4122-8ae1-a938719a623a">